control-plane/pkg/reconciler/channel/v2/channelv2.go

```diff
@@ -779,3 +814,50 @@ func (r *Reconciler) getCaCerts() (*string, error) {
 	}
 	return pointer.String(string(caCerts)), nil
 }
+
+func (r *Reconciler) setTrustBundles(ct *contract.Contract) (bool, error) {
+	tb, err := coreconfig.TrustBundles(r.ConfigMapLister.ConfigMaps(r.SystemNamespace))
+	if err != nil {
+		return false, fmt.Errorf("failed to get trust bundles: %w", err)
+	}
+
+	changed := false
+	if !equality.Semantic.DeepEqual(tb, ct.TrustBundles) {
+		changed = true
+	}
+
+	ct.TrustBundles = tb
+	return changed, nil
+}
+
+// d1 takes preference over d22, where both are set
+func mergeDeliverySpecs(d1, d2 *v1.DeliverySpec) *v1.DeliverySpec {
```

Comment:
Is this the same behavior we used to have on v1 ?

Reply:
I _think_ so - I adapted it from https://github.com/knative-extensions/eventing-kafka-broker/blob/49ad678b4a629be09ad40f9bee1a7c360837102a/control-plane/pkg/core/config/utils.go#L181 which is used in v1.